runtime/lua/vim/lsp/handlers.lua

```diff
@@ -196,7 +196,7 @@ M[ms.workspace_configuration] = function(_, result, ctx)
 
   local response = {}
   for _, item in ipairs(result.items) do
-    if item.section then
+    if item.section and item.section ~= vim.NIL then
```

Comment:
I don't think that we need to check for `~= vim.NIL`, since `section: nil` isn't allowed by the protocol.

Reply:
It is indeed unnecessary for spec compliance. I was just trying to be a bit more robust the face of wonky JSON serialization.

Reply:
I don't think such robustness is needed ;) Besides, this check would support misbehaving clients.

Reply:
IIRC, we will delete JSON keys that are `null` in dictinonaries ( due to ` pcall(vim.json.decode, body, { luanil = { object = true } }` in `lsp/rpc.lua` ), so we don't need to check for `vim.NIL` explicitly regardless.